### PR TITLE
feat: subscription component dashboard implementation

### PR DESCRIPTION
**Pull request by kibo-nehabhanushali**

### CODE REVIEW

File: lib/gql/fragments/subscription.ts [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    weightedOrderAdjustment
+    weightedOrderDiscount
+    adjustedLineItemSubtotal
+    totalWithoutWeightedShippingAndHandling
+    weightedOrderTax
+    weightedOrderShipping
+    weightedOrderShippingDiscount
+    weightedOrderShippingManualAdjustment
+    weightedOrderShippingTax
+    weightedOrderHandlingFee
```
Comment (kibo-kevinwatts) [inline lib/gql/fragments/subscription.ts:75]: Are we using all of these fields?

File: lib/gql/fragments/subscription.ts [hunk 2]
```diff
@@ -0,0 +1,256 @@
+export const baseSubscriptionFragment = /* GraphQL */ `
+  fragment baseSubscriptionFragment on Subscription {
```
Comment (kibo-kevinwatts) [inline lib/gql/fragments/subscription.ts]: I don't see this "baseSubscriptionFragment" used in any queries. What it is this for?

File: hooks/queries/subscription/useSubscriptionsQueries/useSubscriptionsQueries.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  const client = makeGraphQLClient()
+  const response = await client.request({
+    document: getSubscriptionsQuery,
+    variables: {},
+  })
+  return response?.subscriptions
+}
+
+export const useSubscriptionsQueries = (): SubscriptionType => {
+  try {
```
Comment (kibo-kevinwatts) [inline hooks/queries/subscription/useSubscriptionsQueries/useSubscriptionsQueries.ts]: I don't think we are using this try catch pattern around any of the other hooks

File: components/my-account/Subscription/Subscription.tsx [hunk 4]
```diff
@@ -1,164 +0,0 @@
-import React from 'react'
-
-import { Card, Stack, Typography, CardContent, Button } from '@mui/material'
-import { useTranslation } from 'next-i18next'
-
-import { ProductItem } from '@/components/common'
-import { subscriptionGetters, productGetters } from '@/lib/getters'
-import { uiHelpers } from '@/lib/helpers'
-
```
Comment (kibo-kevinwatts) [inline components/my-account/Subscription/Subscription.tsx:9]: Why modify the existing single Subscription component to now handle a list of subscriptions?
We want to keep the single Subscription Component

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: components/my-account/Subscription/MySubscription.tsx [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  const { data: subscriptionDetails } = useSubscriptionsQueries()
+
+  const { getProductLink } = uiHelpers()
+  const { t } = useTranslation('common')
+
+  const handleAccountTitleClick = () => onAccountTitleClick()
+
+  return (
+    <>
+      <Stack sx={{ mt: '2%' }}>
```
Comment (TriptiVermaXP) [inline components/my-account/Subscription/MySubscription.tsx]: We can directly use few props like margin, padding, fonts etc to Typography and stack tags, instead applying sx everywhere.

File: components/my-account/Subscription/MySubscription.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+
+import { ComponentStory, ComponentMeta } from '@storybook/react'
+
+import MySubscription from './MySubscription'
+
+export default {
+  title: 'Subscription/Subscription',
```
Comment (TriptiVermaXP) [inline components/my-account/Subscription/MySubscription.stories.tsx]: component name and story book title mismatched

File: lib/gql/fragments/subscription.ts (hunk 1 above)
Comment (kibo-nehabhanushali) [reply lib/gql/fragments/subscription.ts:75]: No , Removed it

File: hooks/queries/subscription/useSubscriptionsQueries/useSubscriptionsQueries.ts (hunk 3 above)
Comment (kibo-nehabhanushali) [reply hooks/queries/subscription/useSubscriptionsQueries/useSubscriptionsQueries.ts]: sure, removed it

File: components/my-account/Subscription/MySubscription.tsx [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+              <Stack direction="column" sx={{ ...style.subscriptionNumber }}>
+                <Stack direction="row">
+                  <Typography variant="h4" align="right" sx={{ fontWeight: 'bold' }}>
+                    {t('subscription-number')}
+                  </Typography>
+                  <Typography variant="h4" align="right">
+                    {subscriptionGetters.getSubscriptionNumber(
+                      subscriptionItemData as Subscription
+                    )}
+                  </Typography>
```
Comment (TriptiVermaXP) [inline components/my-account/Subscription/MySubscription.tsx]: can we use ProductOption component instead typography for key pair value.
ex:
 <ProductOption option={{ name: t('subscription-number'), value: subscriptionGetters.getSubscriptionNumber(subscriptionItemData as Subscription)}} variant="body2" />

File: components/my-account/Subscription/MySubscription.tsx [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+                    {subscriptionGetters.getSubscriptionFrequency(
+                      subscriptionItemData as Subscription
+                    )}
+                  </Typography>
+                </Stack>
+                <Stack direction="row">
+                  <Typography>{t('estimated-next-arrival-date')}</Typography>
+                  <Typography sx={{ fontWeight: 'bold', pl: '10px' }}>
+                    {subscriptionGetters.nextOrderItemDate(subscriptionItemData as Subscription)}
+                  </Typography>
```
Comment (TriptiVermaXP) [inline components/my-account/Subscription/MySubscription.tsx]: unit tests are missing for all subscription values.

File: components/page-templates/MyAccountTemplate/MyAccountTemplate.tsx [hunk 9]
```diff
@@ -161,6 +165,21 @@ const MyAccountTemplate = () => {
...
+        <Box
+          sx={{
+            ...style.myAccountChildren,
+            ...style.orderHistory,
+          }}
+          onClick={handleGoToSubscription}
+        >
+          <Typography variant="h3">{t('my-subscription')}</Typography>
+          <ChevronRightIcon />
+        </Box>
```
Comment (TriptiVermaXP) [inline components/page-templates/MyAccountTemplate/MyAccountTemplate.tsx]: unit test is missing

**Review by TriptiVermaXP — CHANGES_REQUESTED** [review]
Please check review comments

File: lib/gql/fragments/subscription.ts (hunk 2 above)
Comment (kibo-nehabhanushali) [reply lib/gql/fragments/subscription.ts]: It was added for base but as there are only 2 properties we need from this , removed this fragment

File: components/my-account/Subscription/Subscription.tsx (hunk 4 above)
Comment (kibo-nehabhanushali) [reply components/my-account/Subscription/Subscription.tsx:9]: Done , separeted subscription item and components

File: components/my-account/Subscription/MySubscription.stories.tsx (hunk 6 above)
Comment (kibo-nehabhanushali) [reply components/my-account/Subscription/MySubscription.stories.tsx]: done , updated

File: components/my-account/Subscription/MySubscription.tsx (hunk 8 above)
Comment (kibo-nehabhanushali) [reply components/my-account/Subscription/MySubscription.tsx]: unit test cases are remaining , work is in progress

File: components/my-account/Subscription/MySubscription.tsx (hunk 5 above)
Comment (kibo-nehabhanushali) [reply components/my-account/Subscription/MySubscription.tsx]: done

File: components/my-account/Subscription/MySubscription.tsx (hunk 7 above)
Comment (kibo-nehabhanushali) [reply components/my-account/Subscription/MySubscription.tsx]: done

**Review by TriptiVermaXP — APPROVED** [review]

File: components/my-account/Subscription/MySubscription.tsx (hunk 8 above)
Comment (kibo-nehabhanushali) [reply components/my-account/Subscription/MySubscription.tsx]: done

File: components/page-templates/MyAccountTemplate/MyAccountTemplate.tsx (hunk 9 above)
Comment (kibo-nehabhanushali) [reply components/page-templates/MyAccountTemplate/MyAccountTemplate.tsx]: done

File: components/my-account/Subscription/SubscriptionList/SubscriptionList.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+NoSubscription.parameters = {
+  msw: {
+    handlers: {
+      subscriptions: graphql.query('getSubscriptions', (_req, res, ctx) => {
+        return res(ctx.data(noSubscriptionMock))
+      }),
+    },
+  },
+}
```
Comment (kibo-sushant) [inline components/my-account/Subscription/SubscriptionList/SubscriptionList.stories.tsx:34]: I think, storybook should show multiple subscriptions

File: components/my-account/Subscription/SubscriptionItem/SubscriptionItem.spec.tsx [hunk 11]
```diff
@@ -6,7 +6,7 @@ import { composeStories } from '@storybook/testing-react'
 import { render, screen } from '@testing-library/react'
 import userEvent from '@testing-library/user-event'
 
-import * as stories from './Subscription.stories' // import all stories from the stories file
+import * as stories from './SubscriptionItem.stories' // import all stories from the stories file
```
Comment (kibo-sushant) [inline components/my-account/Subscription/SubscriptionItem/SubscriptionItem.spec.tsx:8]: Should render component - not verifying whether following things are rendered or not

SubscriberName
SubscriberAddress
Edit Frequency
Edit Order Date
Cancel An Item
Edit Billing Information
Edit Shipping Address
Pause Subscription

File: components/my-account/Subscription/SubscriptionItem/SubscriptionItem.tsx [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+const SubscriptionButton = (props: SubscriptionButtonProps) => {
+  const { subscriptionButtonName } = props
+  const { t } = useTranslation('common')
+
+  return (
+    <Button variant="contained" color="secondary" sx={{ ...style.button }}>
+      {t(subscriptionButtonName)}
+    </Button>
+  )
+}
```
Comment (kibo-sushant) [inline components/my-account/Subscription/SubscriptionItem/SubscriptionItem.tsx:76]: Do we need this as a separate component?

File: components/my-account/Subscription/SubscriptionList/SubscriptionList.spec.tsx [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+import * as stories from './SubscriptionList.stories' // import all stories from the stories file
+import { server } from '@/__mocks__/msw/server'
+import { noSubscriptionMock } from '@/__mocks__/stories/subscriptionCollectionMock'
+
+jest.mock('../SubscriptionItem/SubscriptionItem.tsx', () => ({
+  __esModule: true,
+  default: ({ children }: { children: ReactNode }) => (
+    <div data-testid="subscription-item-list-mock">{children}</div>
+  ),
+}))
```
Comment (kibo-sushant) [inline components/my-account/Subscription/SubscriptionList/SubscriptionList.spec.tsx:19]: we are not passing any children in actual component, hence its not required
change subscription-item-list-mock to subscription-item-mock

File: components/my-account/Subscription/SubscriptionList/SubscriptionList.spec.tsx [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+
+    expect(subscriptionTitle).toBeVisible()
+    expect(accountTitleText).toBeVisible()
+
+    await waitFor(async () => {
+      const subscriptionItemList = screen.getByTestId('subscription-item-list-mock')
+
+      await waitFor(async () => expect(subscriptionItemList).toBeVisible())
+    })
+  })
```
Comment (kibo-sushant) [inline components/my-account/Subscription/SubscriptionList/SubscriptionList.spec.tsx:56]: actually we are calling subscriptionItem in loop, so we should check whether same number of subscription-item-mock are rendered or not

**Review by TriptiVermaXP — APPROVED** [review]

File: components/my-account/Subscription/SubscriptionList/SubscriptionList.spec.tsx (hunk 14 above)
Comment (TriptiVermaXP) [reply components/my-account/Subscription/SubscriptionList/SubscriptionList.spec.tsx:56]: Done

File: components/my-account/Subscription/SubscriptionList/SubscriptionList.spec.tsx (hunk 13 above)
Comment (TriptiVermaXP) [reply components/my-account/Subscription/SubscriptionList/SubscriptionList.spec.tsx:19]: Done

File: components/my-account/Subscription/SubscriptionItem/SubscriptionItem.tsx (hunk 12 above)
Comment (TriptiVermaXP) [reply components/my-account/Subscription/SubscriptionItem/SubscriptionItem.tsx:76]: Done

**Review by kibo-sushant — APPROVED** [review]

File: components/my-account/Subscription/SubscriptionItem/SubscriptionItem.spec.tsx (hunk 11 above)
Comment (TriptiVermaXP) [reply components/my-account/Subscription/SubscriptionItem/SubscriptionItem.spec.tsx:8]: Done

File: components/my-account/Subscription/SubscriptionList/SubscriptionList.stories.tsx (hunk 10 above)
Comment (TriptiVermaXP) [reply components/my-account/Subscription/SubscriptionList/SubscriptionList.stories.tsx:34]: Done

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**